File: services/crawler/src/model/news_discover_result.py
from pydantic import BaseModel, ConfigDict, Field


class DiscoveredNewsLink(BaseModel):
    title: str | None = None
    url: str


class NewsDiscoverResult(BaseModel):
    model_config = ConfigDict(populate_by_name=True)

    source_url: str = Field(alias="sourceUrl")
    final_url: str = Field(alias="finalUrl")
    status_code: int = Field(alias="statusCode")
    source_name: str = Field(alias="sourceName")
    source_type: str = Field(alias="sourceType")
    total_found: int = Field(alias="totalFound")
    links: list[DiscoveredNewsLink]

File: backend/src/main/java/com/bupt/publicopinion/collection/vo/NewsDiscoverResult.java
package com.bupt.publicopinion.collection.vo;

import java.util.List;

public record NewsDiscoverResult(
        String sourceUrl,
        String finalUrl,
        int statusCode,
        String sourceName,
        String sourceType,
        int totalFound,
        List<DiscoveredNewsLink> links
) {
}
